test/run-drun/map-upgrades/map0.mo

```diff
@@ -0,0 +1,82 @@
+import Prim "mo:⛔";
+import Cycles = "../cycles/cycles";
+import Lib "node0";
+
+// A naive, distributed map from Nat to Text.
+// Illustrates dynamic installation of imported actor classes.
+// Uses a fixed number of nodes, dynamically installed on demand 
+// .. and upgraded with a call to upgradeNodes() (without data loss)
+
+actor a {
+
+  type Key = Nat;
+  type Value = Text;
+
+  // Number of Nodes
+  let n = 8;
+
+  type Node = Lib.Node;
+
+  stable let savedNodes : [var ?(actor{})] = Prim.Array_init(n, null);
+
+  let nodes : [var ?Node] = Prim.Array_init(n, null);
```

Comment:
Why not making `nodes` stable directly?

Reply:
It's mutable, so invariant, and I expect you wouldn't be able to upgrade to a richer Node type. Using the less informative savedNodes works across evolution of the Node type.

But I could be wrong...